src/context/user.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { createContext, Dispatch, useReducer } from 'react';
+import { produce } from 'immer';
+
+type Token = {
+	symbol: string;
+	name: string;
+	address: string;
+};
+
+type Asset = {
+	id: string;
+	token: Token;
+	value: string;
+};
+
+type Wallet = {
+	assets: Asset[];
+};
+export enum TransactionType {
+	MINT = 'MINT',
+	SWAP = 'SWAP',
+}
+export type Transaction = {
+	id: string;
+	url: '';
+	type: TransactionType;
+	displayName: string;
+	executedOn: string;
+	incomingToken: Token;
+	outgoingToken?: Token;
+	value: string;
+};
+
+export enum ActionTypes {
+	ADD_TRANSACTION = 'ADD_TRANSACTION',
+	UPDATE_ASSET = 'UPDATE_ASSET',
+	ADD_ASSET = 'ADD_ASSET',
```

Comment:
We probably don't need this given that we're using react query and assuming that the balance will be available in the API, but that's a change that we can do later on.